### ISSUE TITLE
Spread operator incorrect type generation

### ISSUE THREAD

**Issue by zane-atomus** [supabase/supabase-js#1253]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When joining a related one-to-one table using the `...` spread operator, the type generation assumes that the rows exist in the related table and does not have `null` as an option for the child table column values, however the returned data has `null` values for rows where the row is missing.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Create a table `parent` with an `id` column as the primary key.
2. Create a table (child) with a foreign key relationship to `parent (id)` (we'll call this column `parent_id`), and a constraint `unique (parent_id)`. For the example we will add two columns `col_1`, `col_2` which have type `boolean` and are `not null`.
3. Run a query with `supabase-js` with the spread operator `const { data, error } = db.from("parent").select("id,...child(col_1,col_2)`.

## Expected behavior

Since the `child` row might not exist for a given `parent` row, the type of `data` should be:
```
{
    id: string;
    col_1: boolean | null;
    col_2: boolean | null;
}
```
instead it gives:
```
{
    id: string;
    col_1: boolean;
    col_2: boolean;
}
```
The returned data contains `null` values as expected which makes the typing inaccurate.

## System information

- OS: macOS
- Version of supabase-js: 2.44.1
- Version of Node.js: 20.12.1

**Comment by avallete** [comment]
Hey there !

We've reworked the result inference in `postgrest-js` to address the typing issues mentioned above. The changes are available in a canary release, and we're actively seeking your feedback.

To test it out, update your `supabase-js` to version `2.46.0-rc.1` (which includes `postgrest-js` `1.17.0-rc.1`):

```bash
npm install supabase-js@2.46.0-rc.1
```

**Comment by zane-atomus** [comment]
> Hey there !
> 
> We've reworked the result inference in `postgrest-js` to address the typing issues mentioned above. The changes are available in a canary release, and we're actively seeking your feedback.
> 
> To test it out, update your `supabase-js` to version `2.46.0-rc.1` (which includes `postgrest-js` `1.17.0-rc.1`):
> 
> ```shell
> npm install supabase-js@2.46.0-rc.1
> ```

Hi,

From our initial testing, our queries appear to be generating the expected return types now! Looking forward to the release of this version.

**Comment by avallete** [comment]
The fix has been released in `supabase-js` v2.46.0. I'm closing this issue, but feel free to reopen if you encounter any further errors.
